syntax = "proto3";

service ClassCall {
  rpc AddClass(AddClassRequest) returns (AddClassReply){}
  rpc RemoveClass(RemoveClassRequest) returns (RemoveClassReply){}
  rpc AddUserToClass(AddUserToClassRequest) returns (AddUserToClassReply){}
  rpc RemoveUserFromClass(RemoveUserFromClassRequest) returns (RemoveUserFromClassReply){}
  rpc GetAllUsersFromClass(GetAllUsersFromClassRequest) returns (GetAllUsersFromClassReply) {}
}

message AddClassRequest {
  string teacher_username = 1;
  string class_name = 2;
  int32 hour = 3;
  int32 room_id = 4;
}

message AddClassReply {
  optional string message = 1;
  optional string error = 2;
  int32 status_code = 3;
}

message RemoveClassRequest {
  int32 class_id = 1;
}

message RemoveClassReply {
  optional string message = 1;
  optional string error = 2;
  int32 status_code = 3;
}

message AddUserToClassRequest {
  int32 class_id = 1;
  string username = 2;
}

message AddUserToClassReply {
  optional string message = 1;
  optional string error = 2;
  int32 status_code = 3;
}

message RemoveUserFromClassRequest {
  int32 class_id = 1;
  string username = 2;
}

message RemoveUserFromClassReply {
  optional string message = 1;
  optional string error = 2;
  int32 status_code = 3;
}

message GetAllUsersFromClassRequest {
  int32 class_id = 1;
}

message GetAllUsersFromClassReply {
  optional string message = 1;
  optional string error = 2;
  int32 status_code = 3;
}